app/actions/views/channel.js

```diff
@@ -373,6 +374,21 @@ export function handleSelectChannel(channelId) {
     };
 }
 
+export function handleSelectChannelByName(channelName, teamName) {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const {teams: currentTeams, currentTeamId} = state.entities.teams;
+        const currentTeamName = currentTeams[currentTeamId].name;
+        const {data: channel} = await getChannelByNameAndTeamName(teamName || currentTeamName, channelName)(dispatch, getState);
```

Comment:
Would it make sense to check if we already have the channel before calling `getChannelByNameAndTeamName`?